CONTRIBUTING.md

```diff
@@ -0,0 +1,261 @@
+
+### Unit Tests
+
+| Action | Command | Required? |
+|--------|---------|-----------|
+| Run all unit tests | `make test` | ✅ Yes |
+| Run with coverage | `make testcov` | ✅ Yes |
+| CI must pass | Automatic | ✅ Yes |
+
+**Requirements**: 
+- All PRs must include unit tests for new code
+- New code must have minimum 80% test coverage ([SonarCloud](https://sonarcloud.io/project/overview?id=ansible_terraform-provider-aap))
+
+Run `make testcov` to generate coverage report locally at `./unit-testing.cov`.
+
+### Acceptance Tests
+
+| Action | Command | Required? |
+|--------|---------|-----------|
+| Setup test resources | `ansible-playbook testing/playbook.yml` | First time only |
+| Source test vars | `source testing/acceptance_test_vars.env` | Before each run |
+| Run acceptance tests | `make testacc` | ✅ Yes |
+
+**Requirement**: PRs must include acceptance test output in PR description.
+
+#### Environment Setup
+
+```bash
+# Set AAP credentials
+export AAP_USERNAME=admin
+export AAP_PASSWORD=your_password
+export AAP_HOSTNAME=http://localhost:9080
+export AAP_INSECURE_SKIP_VERIFY=true
+export AAP_VALIDATE_CERTS=false
+
+# For AAP 2.4
+export CONTROLLER_OPTIONAL_API_URLPATTERN_PREFIX="/api/"
+
+# Install ansible collection
+ansible-galaxy collection install -r testing/requirements.yml
+
+# Create test resources
+ansible-playbook testing/playbook.yml
+```
+
+⚠️ **WARNING**: Acceptance tests modify real AAP resources. Use a dedicated test instance.
+
+### Linting
+
+| Action | Command | Required? |
+|--------|---------|-----------|
+| Run linters | `make lint` | ✅ Yes |
+| CI must pass | Automatic | ✅ Yes |
+
+**Requirement**: All linting errors must be fixed before merge.
+
+## Code Standards
+
+### Pull Request Checklist
+
+- [ ] Unit tests added/updated
+- [ ] Code coverage ≥80% for new code (`make testcov`)
+- [ ] Acceptance tests pass (include output in PR)
+- [ ] Linting passes (`make lint`)
+- [ ] Documentation updated (if applicable)
+- [ ] Changelog entry added (if applicable)
+- [ ] Examples updated (if applicable)
+- [ ] AI attribution noted (if substantial AI assistance used)
+- [ ] PR is focused on single feature/fix
+
+### Focused PRs
+
+**Keep PRs focused on one feature or fix.** If you find yourself making multiple unrelated changes, split them into separate PRs.
+
+**Good examples:**
+- ✅ New resource + related changes to same resource
+- ✅ Bug fix across multiple related resources
+- ✅ Documentation updates for one feature
+
+**Should be split:**
+- ❌ New feature + unrelated bug fixes
+- ❌ Changes to multiple unrelated resources
+- ❌ Refactoring + new features
+
+**Why?** Focused PRs are easier to review, test, and revert if needed.
+
+### New Resources/Data Sources
+
+| Requirement | Location | Notes |
+|-------------|----------|-------|
+| Implementation | `internal/provider/` | Use Plugin Framework |
+| Unit tests | `*_test.go` | Test all CRUD operations |
+| Acceptance test | `*_test.go` | Use `TF_ACC=1` |
```

Comment:
I think we could use more detail in the descriptions here. Unit tests will need to do more than CRUD to get to 80% and there's more to acceptance tests than `TF_ACC=1`